src/state/src/State.c

```diff
@@ -54,6 +54,55 @@ STATUS createStateMachine(PStateMachineState pStates, UINT32 stateCount, UINT64
     return retStatus;
 }
 
+STATUS createStateMachineWithName(PStateMachineState pStates, UINT32 stateCount, UINT64 customData, GetCurrentTimeFunc getCurrentTimeFunc,
+                                  UINT64 getCurrentTimeFuncCustomData, PCHAR pStateMachineName, PStateMachine* ppStateMachine)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PStateMachineImpl pStateMachine = NULL;
+    UINT32 allocationSize = 0;
+
+    CHK(pStates != NULL && ppStateMachine != NULL && getCurrentTimeFunc != NULL && pStateMachineName != NULL, STATUS_NULL_ARG);
+    CHK(stateCount > 0, STATUS_INVALID_ARG);
+    CHK(STRLEN(pStateMachineName) <= MAX_STATE_MACHINE_NAME_LENGTH, STATUS_STATE_MACHINE_NAME_LEN);
```

Comment:
May want to use STRNLEN for additional safety, like this:
```
CHK(STRNLEN(formatStr, MAX_TIMESTAMP_FORMAT_STR_LEN + 1) <= MAX_TIMESTAMP_FORMAT_STR_LEN, STATUS_MAX_TIMESTAMP_FORMAT_STR_LEN_EXCEEDED);
```

Reply:
Yeah we absolutely need this check since this is client supplied and we've only checked it isn't null... it could be non null terminated at all and result in a memory access violation with STRLEN.